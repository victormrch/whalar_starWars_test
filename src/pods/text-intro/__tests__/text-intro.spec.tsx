import { TextIntroComponent } from '../index';
import { render, screen } from '@testing-library/react';

describe('TextIntro component specs', () => {
	it('should render correctly and create snapshot', () => {
		const { asFragment } = render(<TextIntroComponent />);
		expect(asFragment()).toMatchSnapshot();
	});
	it('should display the title and description container', () => {
		render(<TextIntroComponent />);

		const descriptionText = `This is Whalar's technical test for the position of Frontend Developer. The young padawan, Victor Ramon, will demonstrate his skills in front of the senate to achieve his dream of joining the Jedi order of Whalar. His lightsaber forged with React, Redux, Typescript, Styled components and packaged with Vite will help him on his way to dazzle the great Jedi masters.`;

		const title = screen.getByRole('heading', {
			name: /A long time ago in a galaxy far, far away.../i,
		});

		const description = screen.getByRole('heading', {
			name: descriptionText,
		});

		expect(description).not.toBeNull();

		expect(title).not.toBeNull();
		expect(title).toBeInTheDocument();
		expect(title.tagName).toEqual('H1');
		expect(title.textContent).toBe(
			'A long time ago in a galaxy far,  far away...'
		);
	});
});
