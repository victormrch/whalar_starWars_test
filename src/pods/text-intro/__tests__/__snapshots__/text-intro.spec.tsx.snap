// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextIntro component specs should render correctly and create snapshot 1`] = `
<DocumentFragment>
  <h1
    class="sc-bczRLJ fczBkI"
  >
    A long time ago in a galaxy far, 
    <br />
     far away...
  </h1>
  <div
    class="sc-gsnTZi jNwNDw"
  >
    <h1
      class="sc-dkzDqf ixBrQW"
    >
      This is Whalar's technical test for the position of Frontend Developer. 
      <br />
      The young padawan, Victor Ramon, will demonstrate his skills in front of the senate to achieve his dream of joining the Jedi order of Whalar. 
      <br />
      His lightsaber forged with React, Redux, Typescript, Styled components and packaged with Vite will help him on his way to dazzle the great Jedi masters.
    </h1>
  </div>
</DocumentFragment>
`;
